source/guides/guides/test-retries.md

```diff
@@ -0,0 +1,210 @@
+
+### Configure specific retries for different modes
+
+If you want to define different retry attempts for `cypress run` versus `cypress open`, you can pass the `retries` option an object with the following options:
+
+- `runMode` allows you to define the number of test retries when running `cypress run`
+- `openMode` allows you to define the number of test retries when running `cypress open`
+
+```jsx
+{
+  "retries": {
+    // Configure retries for `cypress run`
+    // Default is 0
+    "runMode": 2,
+    // Configure retries for `cypress open`
+    // Default is 0
+    "openMode": 1
+  }
+}
+```
+
+## Custom Configurations
+
+### Individual Test(s)
+
+If you want a specific test to run a custom number of retries, you can set this by using the {% url "test's configuration" writing-and-organizing-tests#Test-Configuration %}.
+
+```jsx
+// Customize retries for an individual test
+describe('User sign-up and login', () => {
+  // `it` test block with no custom configuration
+  it('should redirect unauthenticated user to sign-in page', () => {
+    // ...
+  })
+
+  // `it` test block with custom configuration
+  it('allows user to login', {
+    retries: {
+      runMode: 2,
+      openMode: 1
+    }
+  }, () => {
+    // ...
+  })
+})
+```
+
+### Test Suite(s)
+
+If you want a suite of tests to re-run a custom number of times, you can do this by setting the suite's configuration.
+
+```jsx
+// Customizing retries for a suite of tests
+describe('User bank accounts', {
+  retries: {
+    runMode: 2,
+    openMode: 1,
+  }
+}, () => {
+  // Individual tests will be run normally
+  it('allows a user to view their transactions, () => {
+    // ...
+  }
+
+  it('allows a user to edit their transactions, () => {
+    // ...
+  }
+})
+```
+
+You can find more information about custom configurations here: {% url "Test Configuration" configuration#Test-Configuration %}
+
+# Screenshots
+
+With test retries, Cypress will take a separate screenshot for each test attempt. These will be properly labeled with the attempt number appended at the end of each filename.
+
+With the following test code, you would see the below screenshot filename when all 3 attempts fail:
+
+```js
+describe('User Login', () => {
+  it('displays login errors', () => {
+    cy.visit('/')
+    cy.screenshot()
+    // ...
+  })
+})
+```
+
+```js
+// screenshot filename from cy.screenshot() on 1st attempt
+'User Login -- displays login errors.png'
+// screenshot filename on 1st failed attempt
+'User Login -- displays login errors (failed).png'
+// screenshot filename from cy.screenshot() on 2nd attempt
+'User Login -- displays login errors (attempt 2).png'
+// screenshot filename on 2nd failed attempt
+'User Login -- displays login errors (failed) (attempt 2).png'
+// screenshot filename from cy.screenshot() on 3rd attempt
+'User Login -- displays login errors (attempt 3).png'
+// screenshot filename on 3rd failed attempt
+'User Login -- displays login errors (failed) (attempt 3).png'
```

Comment:
```suggestion
// screenshot filename from cy.screenshot() on 1st attempt
'my-image.png'
// screenshot filename on 1st failed attempt
'User Login -- displays login errors (failed).png'
// screenshot filename from cy.screenshot() on 2nd attempt
'my-image (attempt 2).png'
// screenshot filename on 2nd failed attempt
'User Login -- displays login errors (failed) (attempt 2).png'
// screenshot filename from cy.screenshot() on 3rd attempt
'my-image (attempt 3).png'
// screenshot filename on 3rd failed attempt
'User Login -- displays login errors (failed) (attempt 3).png'
```

Reply:
Yah, I thought about this, but...I guess I maybe want people to see the exact behavior if they don't pass any argument to `cy.screenshot()`? I know this makes it more confusing but...don't they kind of need to understand this?

Reply:
This is an interesting point for sure. I think in this case, it would probably be nice to surface that user's have this ability to name their screenshots. Users who aren't used to doing this should be able to follow this given the simplicity of the example, so I think there is merit to @Bkucera's change.